service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -118,6 +90,42 @@ public void prepareCustomDirections(CaseDetails caseDetails, CaseManagementOrder
         }
     }
 
+    public DynamicList getHearingDateDynamicList(List<Element<HearingBooking>> hearings, CaseManagementOrder order) {
+        List<DynamicListElement> values = getDateElements(hearings);
+
+        DynamicListElement selectedValue = ofNullable(order)
+            .map(x -> getPreselectedDate(values, x.getId()))
+            .orElse(DynamicListElement.EMPTY);
+
+        return DynamicList.builder()
+            .listItems(values)
+            .value(selectedValue)
+            .build();
+    }
+
+    //TODO: isAfter method in hearing booking does not filter out todays date as Time is set before LocalDate.now()
+    // is evaluated
+    private List<DynamicListElement> getDateElements(List<Element<HearingBooking>> hearings) {
+        return hearings.stream()
+            .filter(hearingBooking -> hearingBooking.getValue().getStartDate().isAfter(time.now()))
+            .map(this::buildDynamicListElement)
+            .collect(toList());
+    }
+
+    private DynamicListElement buildDynamicListElement(Element<HearingBooking> element) {
+        return DynamicListElement.builder()
+            .label(formatLocalDateToString(element.getValue().getStartDate().toLocalDate(), FormatStyle.MEDIUM))
+            .code(element.getId())
+            .build();
+    }
+
+    private DynamicListElement getPreselectedDate(List<DynamicListElement> list, UUID id) {
+        return list.stream()
+            .filter(item -> item.getCode().equals(id))
+            .findFirst()
+            .orElse(DynamicListElement.EMPTY);
```

Comment:
It seems we are doing that or else EMPTY twice? maybe this could return Optional? Will the other one be invoked?

Reply:
hmm do you not think the two empties are both valid? One in this method is for if there is an id that doesn't match -> which can happen if hearings are deleted (although we don't deal with this currently at all) the other is if order is null. Maybe one empty catches both? I'll take a look

Reply:
Sure makes sense, happy either way.

But at the same time was thinking client could deal with fact there is no preselected date, or something like that :)